### ISSUE TITLE
LaTeX rendering of nice numbers aren't always correct

### ISSUE THREAD

**Issue by DmitryVasilevsky** [microsoft/qsharp#1865]
Useful functions to render some rational, algebraic numbers and exponential form of complex number aren't always producing the expected result when used outside state rendering. Now that we have want to add a few other use cases we should fix that.
Example:
![image](https://github.com/user-attachments/assets/4a2aa1be-dbcb-4b4d-a3b3-e0c75d7a4876)

**Comment by sezna** [comment]
After #1885, these numbers will rarely (never?) show up in output, so we are marking this as low priority.

**Comment by DmitryVasilevsky** [comment]
Code changes in https://github.com/microsoft/qsharp/pull/1885 provide correct handling of standalone numbers (rather than coefficients).  The following can still be improved in general case:
    * i could be used directly in a fraction. Example: $\frac{1}{2}i$ could be rendered as $\frac{i}{2}$.
    * In a simple fraction sign can be moved to numerator: Example: $-\frac{1}{2}$ could be rendered as $\frac{-1}{2}$.
    * Brackets are not needed for standalone rendering. Example: $-\left(1+i\right)$ could be rendered as $-1-i$
    * No need to rationalize the denominator. Example: $\frac{\sqrt{2}}{2}$ could be rendered as $\frac{1}{\sqrt{2}}$
